### PR TITLE
add specific styles for with-dots variant & a bit of clean up #779

### LINKED ISSUE #779: Bug: [V2 product listing] - Padding and Layout Issues
### Issue Type

Bug

### Summary

Incorrect formatting of the product listing for the All Trucks page. When the v2 all trucks block was combined into the v2 product listing the formatting of the former was changed and is now inconsistent with the [Figma design.](https://www.figma.com/design/URJhT8pkntqc6h6v5wSLqx/Mack-Core-Design-Kit?node-id=4901-12110&t=bIL1ZknDTUzXIUU8-1) 

### Priority

Blocker

### Description

The spacing and padding for the text in the v2 product listing block are incorrect, affecting several elements: 
- [ ] The product title doesn't have proper spacing of 135px above
- [ ] The industry tags are displayed vertically instead of horizontally with pipes separating them.
- [ ] The buttons are positioned too far underneath the text, with excessive padding (16px instead of the required amount).
- [ ] Font size for heading is too big, should be H3

Relevant tickets: 
v2 all trucks issue: https://github.com/Netcentric/vg-macktrucks-com-rd/issues/55

v2 product listing issue: https://github.com/hlxsites/vg-macktrucks-com/issues/580

### Affected Device Classes

Mobile, Tablet, Desktop

### Affected Browsers/Devices and Versions

All

### Steps to Reproduce

1. Navigate to test page that has v2-product-list block: https://main--vg-macktrucks-com--hlxsites.hlx.page/trucks/index-new
2. Observe the placement and padding spacing of the product title, industry tags, and buttons which are not consistent with the Figma requirements. 
3. Link to Figma for expected results: https://www.figma.com/design/URJhT8pkntqc6h6v5wSLqx/Mack-Core-Design-Kit?m=dev&node-id=4901-12110

### Expected Results

Image

Image

### Actual Results

Image

### PR DESCRIPTION
**Pull request by jonatan-lledo-netcentric**
# Fix Updates

I split styling for the `default` and the `with-dots` variant

## Notes

In the Library I had to add a disclaimer note to check if the variant has links in each row, if not it has to be added after copying it.
Also, I didn't test it for the `with-filter` variant, so if it is also affected then we will need to raise a new ticket.

#

Fix #779

Test URLs:
- Before: https://main--vg-macktrucks-com--hlxsites.hlx.page/drafts/jlledo/v2/index-new
- After: https://779-product-listing-padding--vg-macktrucks-com--hlxsites.hlx.page/drafts/jlledo/v2/index-new

### CODE REVIEW

**Review by alexiscoelho — APPROVED** [review]

File: blocks/v2-product-listing/v2-product-listing.css [hunk 1]
```diff
@@ -106,25 +106,21 @@
 }
 
 .v2-product-listing__button-container {
-  margin-top: 12px;
+  margin-top: 4px; /* 4 + 12 = 16px to match design */
   width: 100%;
   display: flex;
   gap: 16px;
```
Comment (santi-homps) [inline blocks/v2-product-listing/v2-product-listing.css:112]: shouldnt we use `--v2-product-listing-content-gap` here?

File: blocks/v2-product-listing/v2-product-listing.css [hunk 2]
```diff
@@ -156,33 +152,49 @@
 }
 
 .v2-product-listing__detail-list {
-  text-transform: capitalize;
+  gap: 16px;
```
Comment (santi-homps) [inline blocks/v2-product-listing/v2-product-listing.css:155]: also here use `--v2-product-listing-content-gap`

Comment (jonatan-lledo-netcentric) [reply blocks/v2-product-listing/v2-product-listing.css:155]: done

File: blocks/v2-product-listing/v2-product-listing.css (hunk 1 above)
Comment (jonatan-lledo-netcentric) [reply blocks/v2-product-listing/v2-product-listing.css:112]: done

**Review by santi-homps — APPROVED** [review]

**Pull request merged**